src/go/k8s/pkg/console/configmap.go

```diff
@@ -0,0 +1,310 @@
+	username := string(secret.Data[corev1.BasicAuthUsernameKey])
+	password := string(secret.Data[corev1.BasicAuthPasswordKey])
+
+	config, err := cm.generateConsoleConfig(ctx, username, password)
+	if err != nil {
+		return err
+	}
+
+	// Create new ConfigMap instead of updating existing so Deployment will trigger a reconcile
+	immutable := true
+	obj := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: cm.consoleobj.GetName() + "-",
+			Namespace:    cm.consoleobj.GetNamespace(),
+			Labels:       labels.ForConsole(cm.consoleobj),
+		},
+		Data: map[string]string{
+			"config.yaml": config,
+		},
+		Immutable: &immutable,
+	}
+
+	if err := controllerutil.SetControllerReference(cm.consoleobj, obj, cm.scheme); err != nil {
+		return err
+	}
+	if err := cm.Create(ctx, obj); err != nil {
+		return fmt.Errorf("creating Console configmap: %w", err)
+	}
+
+	// This will get updated in the controller main reconcile function
+	// Other Resources may set Console status if they are also watching GenerationMatchesObserved()
+	cm.consoleobj.Status.ConfigMapRef = &corev1.ObjectReference{Namespace: obj.GetNamespace(), Name: obj.GetName()}
+
+	return nil
+}
+
+// Key implements Resource interface
+func (cm *ConfigMap) Key() types.NamespacedName {
+	return types.NamespacedName{Name: cm.consoleobj.GetName(), Namespace: cm.consoleobj.GetNamespace()}
+}
+
+// generateConsoleConfig returns the actual config passed to Console.
+// This should match the fields at https://github.com/redpanda-data/console/blob/master/docs/config/console.yaml
+// We are copying the fields instead of importing them because (1) they don't have json tags (2) some fields aren't ideal for K8s (e.g. TLS certs shouldn't be file paths but Secret reference)
+func (cm *ConfigMap) generateConsoleConfig(ctx context.Context, username, password string) (string, error) {
+	consoleConfig := &ConsoleConfig{
+		MetricsNamespace: cm.consoleobj.Spec.MetricsNamespace,
+		ServeFrontend:    cm.consoleobj.Spec.ServeFrontend,
+		Server:           cm.consoleobj.Spec.Server,
+		Kafka:            cm.genKafka(username, password),
+	}
+
+	connectConfig, err := cm.genConnect(ctx)
+	if err != nil {
+		return "", err
+	}
+	consoleConfig.Connect = *connectConfig
+
+	config, err := yaml.Marshal(consoleConfig)
+	if err != nil {
+		return "", err
+	}
+	return string(config), nil
+}
+
+var (
+	// Currently issuing TLS certs through LetsEncrypt
+	// SchemaRegistry.TLS.NodeSecretRef is set to Secret without CaFile "ca.crt"
+	// This flag overrides using NodeSecretRef to mount CaFile
+	UsePublicCerts = true
+
+	// Console is creating a NewCertPool() which will not use host default certificates when left blank
+	// REF https://github.com/redpanda-data/console/blob/master/backend/pkg/schema/client.go#L60
+	DefaultCaFilePath = "/etc/ssl/certs/ca-certificates.crt"
+
+	SchemaRegistryTLSDir          = "/redpanda/schema-registry"
+	SchemaRegistryTLSCaFilePath   = fmt.Sprintf("%s/%s", SchemaRegistryTLSDir, "ca.crt")
```

Comment:
I don't recall if ca.crt is also mirrored by cert-manager in case of using a public CA

Reply:
I don't think cert-manager will add CA cert in the Secret. This is only used if `NodeSecretRef` is used [REF](https://github.com/redpanda-data/redpanda/blob/dev/src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go#L665) Otherwise, we are using host CA [REF](https://github.com/redpanda-data/redpanda/pull/5758/files#diff-622d2275ef6540ff881c27df273e8576383c794dbb32d7a868e7265950213735R150)